solr/core/src/java/org/apache/solr/search/TermsQParserPlugin.java

```diff
@@ -142,4 +158,112 @@ public Query parse() throws SyntaxError {
       }
     };
   }
+
+  private static class TopLevelDocValuesTermsQuery extends DocValuesTermsQuery implements ExtendedQuery{
+    private final String fieldName;
+    private SortedSetDocValues topLevelDocValues;
+    private LongBitSet topLevelTermOrdinals;
+    private boolean matchesAtLeastOneTerm = false;
+    private boolean cache = true;
+    private boolean cacheSeparately = false;
+    private int cost;
+
+
+    public TopLevelDocValuesTermsQuery(String field, BytesRef... terms) {
+      super(field, terms);
+      this.fieldName = field;
+    }
+
+    public Weight createWeight(IndexSearcher searcher, final ScoreMode scoreMode, float boost) throws IOException {
+      topLevelDocValues = DocValues.getSortedSet(((SolrIndexSearcher)searcher).getSlowAtomicReader(), fieldName);
```

Comment:
If not a SolrIndexSearcher than return super.createWeight    (your idea based on Mikhail's comment)